sympy/physics/continuum_mechanics/beam.py

```diff
@@ -1116,6 +1117,262 @@ def max_deflection(self):
+        - 2.08333333333333e-5*SingularityFunction(x, 8, 3) + 0.00138541666666667 for x over (0.0, 8.0)
+        """
+        slope = self.slope()
+        for sym in slope.atoms(Symbol):
+            if sym == self.variable:
+                continue
+            if sym not in var:
+                raise ValueError("not enough var ")
+        if self.length in var:
+            length = var[self.length]
+        else:
+            length = self.length
+        return plot(slope.subs(var), (self.variable, 0, length), title='Slope',
+                xlabel='position', ylabel='Value', line_color='m')
+
+    def plot_deflection(self, var={}):
+        """
+        Returns a plot for deflection curve of the Beam object.
+
+        Parameters
+        ==========
+        var : dictionary
+            Python dictionary containing Symbols as key and their
+            corresponding values.
+
+        Examples
+        ========
+        There is a beam of length 8 meters. A constant distributed load of 10 KN/m
+        is applied from half of the beam till the end. There are two simple supports
+        below the beam, one at the starting point and another at the ending point
+        of the beam. A pointload of magnitude 5 KN is also applied from top of the
+        beam, at a distance of 4 meters from the starting point.
+        Take E = 200 GPa and I = 400*(10**-6) meter**4.
+
+        Using the sign convention of downwards forces being positive.
+
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import symbols
+        >>> R1, R2 = symbols('R1, R2')
+        >>> b = Beam(8, 200*(10**9), 400*(10**-6))
+        >>> b.apply_load(5000, 2, -1)
+        >>> b.apply_load(R1, 0, -1)
+        >>> b.apply_load(R2, 8, -1)
+        >>> b.apply_load(10000, 4, 0, end=8)
+        >>> b.bc_deflection = [(0, 0), (8, 0)]
+        >>> b.solve_for_reaction_loads(R1, R2)
+        >>> b.plot_deflection()
+        Plot object containing:
+        [0]: cartesian line: 0.00138541666666667*x - 2.86458333333333e-5*SingularityFunction(x, 0, 3)
+        + 1.04166666666667e-5*SingularityFunction(x, 2, 3) + 5.20833333333333e-6*SingularityFunction(x, 4, 4)
+        - 6.51041666666667e-5*SingularityFunction(x, 8, 3) - 5.20833333333333e-6*SingularityFunction(x, 8, 4)
+        for x over (0.0, 8.0)
+        """
+        deflection = self.deflection()
+        for sym in deflection.atoms(Symbol):
+            if sym == self.variable:
+                continue
+            if sym not in var:
+                raise ValueError('Value of %s was not passed.' %sym)
+        if self.length in var:
+            length = var[self.length]
+        else:
+            length = self.length
+        return plot(deflection.subs(var), (self.variable, 0, length), title='Deflection',
+                xlabel='position', ylabel='Value', line_color='r')
+
+    def plot_all(self, var={}):
```

Comment:
I recommend that this create a subplot since the units are different for all of them.

Reply:
Would simply `plot()` be a sufficient name?

Reply:
One `plot` method already exists in `plotting` module. Is it okay to give same name to this method?

Reply:
I don't see why not. 

But now that I think about this, here we have a "beam". If I were a new user and saw a "plot" method I guess I'd think it would show a diagram of the beam (like what you see in the problem description in engr. books). If I saw "plot_all" I'd probably guess it'd plot multiple things but it isn't that informative about what will be plotted. Maybe "plot" should be reserved for showing a diagram of the beam and "plot_diagrams" or "plot_all" or something better would be for this method. This method you have plots a "shear diagram", "moment diagram", "slope curve", and "deflection curve". Maybe "plot_loaded_results" or "plot_loading_results"...not thinking of a great concise name.